### PR TITLE
multi package support fixes

### PR DESCRIPTION
**Pull request by Himavanth**
## Description
Fixes config.json issue in https://github.com/adobe/aio-lib-runtime/issues/15

#### config.json changes
- Modified keys for default package to be without the package name. (eg:- `action1` instead of `pkg1/action1`)

So for every action in the default package, there will be either of two entries. One is \<packageName\>/\<actionName\> and the other is \<actionName\>
```
{
"package2/hello":"https://23294_62843.adobeio-static.net/api/v1/web/package2/hello",
"hello":"https://23294_62843.adobeio-static.net/api/v1/web/multi-check-0.0.2/hello"
}
```
#### Build zip file changes
- For the default/first package, zip file path is now `dist/<actionName>.zip` instead of `dist/<packageName>-<actionName>.zip` to maintain backward compatibility
- For non-default packages, zip file path would be `dist/<packageName>/<actionName>.zip`. The only case where this could be a problem is when user has an action with name say `action1-temp` and a package with the same name. But it's a remote case.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sandeep-paliwal** [comment]
@Himavanth some files have conflicts
Also there are a lot of JSDoc warnings, although not because of the new changes. we can address those as well.

File: test/utils.test.js [hunk 1]
```diff
@@ -1548,25 +1548,45 @@ describe('getActionUrls', () => {
...
+      'sample-app-1.0.0-action-sequence': 'https://adobeioruntime.net/api/v1/fake_ns/sample-app-1.0.0/action-sequence',
+      'sample-app-1.0.0-action-zip': 'https://adobeioruntime.net/api/v1/web/fake_ns/sample-app-1.0.0/action-zip'
     }
     config.manifest.full.packages.__APP_PACKAGE__.actions.action.web = 'no'
     config.manifest.full.packages.__APP_PACKAGE__.sequences['action-sequence'].web = false
     const result = utils.getActionUrls(config, config.actions.devRemote, true)
     expect(result).toEqual(expect.objectContaining(expected))
   })
+
+  test('multiple packages with includeWithoutPkgName = true', () => {
```
Comment (sandeep-paliwal) [inline test/utils.test.js:1571]: also add test for `multiple packages includeWithoutPackage = false` (default param value for getActionUrls )

**Comment by Himavanth** [comment]
@moritzraho Can you take a look to verify that your changes in the 1.2.1 branch(and master) are intact? With this we can move to master branch and delete the 1.2.1 branch

File: src/deploy-actions.js [hunk 2]
```diff
@@ -72,13 +72,14 @@ async function deployActions (config, deployConfig = {}, logFunc) {
     pkg.version = config.app.version
     for (const [name, action] of Object.entries(pkg.actions)) {
       // change path to built action
-      action.function = path.join(relDist, pkgName + '-' + name + '.zip')
+      const zipFileName = (modifiedConfig.ow.package === pkgName ? name : pkgName + '-' + name) + '.zip'
```
Comment (sandeep-paliwal) [inline src/deploy-actions.js:75]: This logic to create action name based on default package repeats for action build as well, maybe we can move this to utils and use it for build and deploy scripts as required.

**Comment by moritzraho** [comment]
> @moritzraho Can you take a look to verify that your changes in the 1.2.1 branch(and master) are intact? With this we can move to master branch and delete the 1.2.1 branch

Yes I've verified every combination of getActionUrls, things work nicely thanks Hima

**Pull request merged**